### PR TITLE
SI Process

### PR DESCRIPTION
**Pull request by ActuallyTaylor**
Updated HUDl to standard process loop

### CODE REVIEW

File: include/HUDL/HUDL.hpp [hunk 1]
```diff
@@ -45,44 +41,74 @@ class HUDL {
...
-    static constexpr uintptr_t TMS_NODE_ID = 0x08;
-    static constexpr uintptr_t MC_NODE_ID = 0x01;
+    /**
+     * +The internal LCD used for for the HUDL to display.
+     */
+    DEV::LCD lcd;
 
 private:
+    /**
+     * The corner of the display that will content will be displayed in.
```
Comment (mjmagee991) [inline include/HUDL/HUDL.hpp]:
```suggestion
     * The corner of the display that the content will be displayed in.
```

File: include/HUDL/HUDL.hpp [hunk 2]
```diff
@@ -45,44 +41,74 @@ class HUDL {
...
+     * Standard process function for the HUDL. Updates the HUDL display on a set refresh rate.
+     * The refresh rate is defined as REFRESH_RATE
      */
-    DEV::LCD lcd;
+    void process();
 
-    static constexpr uintptr_t TMS_NODE_ID = 0x08;
-    static constexpr uintptr_t MC_NODE_ID = 0x01;
+    /**
+     * +The internal LCD used for for the HUDL to display.
```
Comment (mjmagee991) [inline include/HUDL/HUDL.hpp]:
```suggestion
     * The internal LCD used for for the HUDL to display.
```

File: include/HUDL/HUDL.hpp [hunk 3]
```diff
@@ -91,272 +117,92 @@ class HUDL {
     /** The torque actual value provided by the MC node over CAN. Found in the first 16 bits of the 4th PDO coming from the MC. */
     uint16_t torqueActual = 0;
 
+    /** The current position of the Motor Controller. Used to calculate speed, RPM and more. */
     int16_t actualPosition = 0;
 
+    /**
+     * A static function that retrieves the column number for a specified corner. This is used
```
Comment (mjmagee991) [inline include/HUDL/HUDL.hpp]:
```suggestion
     * Retrieves the column number for a specified corner. This is used
```
You generally don't want to mention things about whatever you're documenting that are present in the thing itself. It just increases your chances of updating the thing and forgetting to update the comment. This includes stuff like static and any data types.

Make sure to fix this in other comments as well

File: include/HUDL/HUDL.hpp [hunk 4]
```diff
@@ -91,272 +117,92 @@ class HUDL {
...
+     * Sets the data to display for a given corner
+     *
+     * @param corner the corner to set data for.
+     * @param text the data to display in the corner.
+     */
     void dataForCorner(Corner corner, const char* text);
 
-    static constexpr uint16_t OBJECT_DICTIONARY_SIZE = 37;
+    /**
+     * The size of the CAN object dictionary.
```
Comment (mjmagee991) [inline include/HUDL/HUDL.hpp]:
```suggestion
     * The size of the CANopen object dictionary.
```

File: include/HUDL/HUDL.hpp [hunk 5]
```diff
@@ -91,272 +117,92 @@ class HUDL {
...
+     */
+    static constexpr uint16_t OBJECT_DICTIONARY_SIZE = 46;
 
+    /**
+     * The CAN object dictionary.
```
Comment (mjmagee991) [inline include/HUDL/HUDL.hpp]:
```suggestion
     * The CANopen object dictionary.
```

File: src/HUDL/HUDL.cpp [hunk 6]
```diff
@@ -140,5 +146,13 @@ uint8_t HUDL::pageForCorner(HUDL::Corner corner) {
         return -1;
     }
 }
+void HUDL::process() {
```
Comment (mjmagee991) [inline src/HUDL/HUDL.cpp:150]:
```suggestion

void HUDL::process() {
```

File: src/HUDL/HUDL.cpp [hunk 7]
```diff
@@ -140,5 +146,13 @@ uint8_t HUDL::pageForCorner(HUDL::Corner corner) {
...
+    if (displayCounter >= REFRESH_RATE) {
+        displayCounter = 0;
 
+        updateLCD();
+    } else {
+        displayCounter++;
+    }
+}
```
Comment (mjmagee991) [inline src/HUDL/HUDL.cpp:158]:
```suggestion
}

```

**Review by mjmagee991 — CHANGES_REQUESTED** [review]
A couple details to clean up

File: include/HUDL/HUDL.hpp [hunk 8]
```diff
@@ -45,44 +41,74 @@ class HUDL {
...
+    uint16_t displayCounter = 0;
+
+    /**
+     * The node IDs used to identify the device on the CAN network.
+     */
+    /** NODE ID for the HUDL */
+    static constexpr uint8_t NODE_ID = 11;
+    /** NODE ID for the TMS */
+    static constexpr uintptr_t TMS_NODE_ID = 8;
+    /** NANO ID for the Motor Controller */
```
Comment (aclowmclaughlin) [inline include/HUDL/HUDL.hpp]:
```suggestion
    /** NODE ID for the Motor Controller */
```

File: include/HUDL/HUDL.hpp [hunk 9]
```diff
@@ -45,44 +41,74 @@ class HUDL {
...
+    static constexpr uintptr_t MC_NODE_ID = 1;
+
+    /** A boolean that tracks whether or not the headers have been set. */
```
Comment (aclowmclaughlin) [inline include/HUDL/HUDL.hpp]:
```suggestion
    /**  Tracks whether or not the headers have been set. */
```

**Review by aclowmclaughlin — CHANGES_REQUESTED** [review]
Looks good except for some minor changes!

**Review by mjmagee991 — APPROVED** [review]
Nice work

**Review by aclowmclaughlin — APPROVED** [review]
"This shit 🔥 "

**Pull request merged**